### PR TITLE
Bug fix in neuron spack recipe

### PR DESCRIPTION
**Pull request by pramodk**
- compiler was not correctly filtered
  - fixes neuronsimulator/nrn/issues/1058

### CODE REVIEW

File: var/spack/repos/builtin/packages/neuron/package.py [hunk 1]
```diff
@@ -103,7 +103,12 @@ def filter_compilers(self):
         nrnmech_makefile = join_path(self.prefix,
                                      "./bin/nrnmech_makefile")
 
-        assign_operator = "?="
+        # assign_operator is changed to fix wheel support
+        if self.spec.satisfies("+cmake") and self.spec.satisfies("@:7.99"):
+            assign_operator = "?="
+        else:
+            assign_operator = "="
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/neuron/package.py]: I don't see any `cmake` variant. So I think this conditional will always take the else branch and is equivalent to:
```suggestion
        assign_operator = "="
```

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/neuron/package.py (hunk 1 above)
Comment (ikitayama) [reply var/spack/repos/builtin/packages/neuron/package.py]: LGTM

**Comment by ikitayama** [comment]
All, I'll check if I can use it correctly once the fix proposed by @pramodk is merged upstream.

File: var/spack/repos/builtin/packages/neuron/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/neuron/package.py]: @alalazo : thank's for catching that. I copied this from our fork where we still have autotools/cmake build.
I fixed the issue : With newer >=8.0 and develop we need this. Hence I just removed `+cmake` from the condition.

**Review by alalazo — APPROVED** [review]

**Comment by ikitayama** [comment]
Do you want to keep this open til the merge happens?

**Comment by alalazo** [comment]
@ikitayama Not sure I understand your comment. I enabled auto-merge, meaning as soon as tests run successfully the PR will be merged. Is there any reason to hold the merge?

**Comment by ikitayama** [comment]
No, please go ahead; I though you closed this Issue that’s what I was
actually wondering

**Comment by alalazo** [comment]
@ikitayama Closed / reopen to restart test cleanly and generate a different merge commit.

**Comment by ikitayama** [comment]
Now I understand.

**Comment by ikitayama** [comment]
I am still seeing this:

```
$ nrnivmodl mod
/p/project/cjzam11/kitayama1/projects/35358
Mod files: "mod/fornetcon.mod" "mod/invlfire.mod" "mod/sample.mod"

Creating x86_64 directory for .o files.

COBJS=''
 -> Compiling mod_func.cpp
 -> NMODL ../mod/invlfire.mod
 -> NMODL ../mod/fornetcon.mod
 -> NMODL ../mod/sample.mod
Spack compiler must be run from Spack! Input 'SPACK_ENV_PATH' is missing.
make: *** [/p/project/cjzam11/kitayama1/projects/spack/opt/spack/linux-centos8-zen2/gcc-10.2.0/neuron-develop-ws2elf3xzhh5eu2kz7wzdneijg62oagc/bin/nrnmech_makefile:120: mod_func.o] Error 1
make: *** Waiting for unfinished jobs....
Translating invlfire.mod into /p/project/cjzam11/kitayama1/projects/35358/x86_64/invlfire.c
Thread Safe
Translating fornetcon.mod into /p/project/cjzam11/kitayama1/projects/35358/x86_64/fornetcon.c
Translating sample.mod into /p/project/cjzam11/kitayama1/projects/35358/x86_64/sample.c
Thread Safe
Thread Safe
```

**Comment by ikitayama** [comment]
I built `neuron` freshly with Spack and loaded it.

**Comment by ikitayama** [comment]
Sorry, I was basing off of `develop` branch; but I suppose this PR isn't merged, yet.

**Comment by ikitayama** [comment]
@pramodk `nrnivmodl mod` works; but `nrnivmodl-core mod` gives:

```
$ nrnivmodl-core mod
[INFO] Running: make -j4 -f /p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile ROOT=/p/project/cjzam11/kitayama1/opt/nrn-master MODS_PATH=x86_64/corenrn/mod2c BUILD_TYPE=SHARED
Default NMODL flags:
diff: x86_64/corenrn/mod2c/_mod_func.cpp: No such file or directory
/p/project/cjzam11/kitayama1/opt/nrn-master/bin/mod2c_core/p/project/cjzam11/kitayama1/opt/nrn-master/bin/mod2c_core: error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory
/p/project/cjzam11/kitayama1/opt/nrn-master/bin/mod2c_core: : error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory
error while loading shared libraries: libiomp5.so: cannot open shared object file: No such file or directory
make: *** [/p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile:225: x86_64/corenrn/mod2c/expsyn.cpp] Error 127
make: *** Waiting for unfinished jobs....
make: *** [/p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile:225: x86_64/corenrn/mod2c/exp2syn.cpp] Error 127
make: *** [/p/project/cjzam11/kitayama1/opt/nrn-master/share/coreneuron/nrnivmodl_core_makefile:225: x86_64/corenrn/mod2c/fornetcon.cpp] Error 127
```

Could you take a look at this on Monday?

**Comment by pramodk** [comment]
@ikitayama : “ libiomp5.so: cannot open shared object file: No such file or directory“ indicates that the compilers modules are not loaded. Could you check if you have loaded necessary modules before launching nrnivmodl-core?

**Comment by ikitayama** [comment]
@pramodk Spack `neuron` doesn't seem to provide `nrnivmodl-core`; and `nrnivmodl -coreneuron` gives:

```
$ nrnivmodl -coreneuron mod
/p/project/cjzam11/kitayama1/projects/35358
Mod files: "mod/fornetcon.mod" "mod/invlfire.mod" "mod/sample.mod"

COBJS=''
ERROR : CoreNEURON support is not enabled!
```

**Comment by ikitayama** [comment]
Sorry, I should've built `neuron` with coreneuron as well. `nrnivmodl-core` is provided and it builds model files fine.

**Comment by pramodk** [comment]
Ok good! I assume build issues are solved now.

**Comment by ikitayama** [comment]
Yes! Thanks for the fix Pramod!

**Pull request merged**